web/client/selectors/geostory.js

```diff
@@ -287,3 +287,9 @@ export const hasPendingChanges = (state = {}) => state?.geostory?.pendingChanges
  * @param {object} state application state
  */
 export const updateUrlOnScrollSelector = state => get(state, 'geostory.updateUrlOnScroll', false);
+
+/**
+ * Gets the currentStoryFonts that were loaded from config by useEffect hook in the GeoStoryEditor
+ * @param {object} state application state
+ */
+export const currentStoryFonts = state => get(state, "geostory.currentStory.settings.fontFamilies", []);
```

Comment:
I tried the configuration proposed here https://github.com/geosolutions-it/MapStore2/issues/5727#issuecomment-671305781 inside the "GeoStoryEditor" but the list of fonts did not showed the new font.

tested configuration:

```js
{
    "name": "GeoStoryEditor",
    "cfg": {
        "disablePluginIf": "{!!(state('browser') && state('browser').mobile)}",
        "containerPosition": "columns",
        "fontFamilies": [
            {
                "family": "Lemonada",
                "src": "https://fonts.googleapis.com/css2?family=Lemonada:wght@300&display=swap"
            }
        ]
    }
}

```

I tried also the above fontFamilies configuration in GeoStory with the same result.

Note: in the screenshot below I changed the font in the editor settings but it's not available in the inline text editor

![image](https://user-images.githubusercontent.com/19175505/91960272-fe96b680-ed09-11ea-943b-c1fb4aaacfa0.png)

After a quick investigation on latest updates (https://github.com/geosolutions-it/MapStore2/pull/5782) I noticed following issues:

- [ ] It seems the `geostory.currentStory.settings.fontFamilies` property it's never updated so this means the currentStoryFonts will return always an empty array. We should decide where save the list of fonts:
  - should we provide them within the [newgeostory.json ](https://github.com/geosolutions-it/MapStore2/blob/b6412bd6dff00e96fc6a12fb66d9929b461aa235/web/client/configs/newgeostory.json#L4-L18)so the selected font family will be related to a specific story?
  - or should we have the possibility to change the list via localConfig.json?

In my opinion this information should be stored in the **newgeostory.json** similar to theme and be persisted in the saved story. In this way we will not have conflict if we decide to change the list of fonts in newgeostory.json because only new story will be affected while the older one will use their own list of fonts. If we want to use localConfig.json we should take into account that probably we should repeat two times the list of font family one for GeoStory and another for GeoStoryEditor or find a way to store this information temporary in the state to make it accessible to both plugins (I don't like this approach)

- [ ] the font loader is located in the [`GeoStoryEditor`](https://github.com/geosolutions-it/MapStore2/blob/b6412bd6dff00e96fc6a12fb66d9929b461aa235/web/client/plugins/GeoStoryEditor.jsx#L108-L123) plugin this means that if we remove this plugin from the localConfig the `GeoStory` plugin will not access to the loaded fonts.

- [ ] Same font is requested multiple times one for each times the GeoStoryEditor is rendered and I'm guessing: could the [fontFamilies](https://github.com/geosolutions-it/MapStore2/blob/b6412bd6dff00e96fc6a12fb66d9929b461aa235/web/client/plugins/GeoStoryEditor.jsx#L123) dependency cause this issue? Anyway we should ensure to not append the same css link multiple times also when mount or unmount a component.
 
![image](https://user-images.githubusercontent.com/19175505/91963403-1cfeb100-ed0e-11ea-85ac-0e0ded5f709a.png)


### Configuration proposal

Move the fontFamilies list from localConfig to [newgeostory.json](https://github.com/geosolutions-it/MapStore2/blob/b6412bd6dff00e96fc6a12fb66d9929b461aa235/web/client/configs/newgeostory.json#L4-L18)

something like 

```js
{
    "type": "cascade",
    "resources": [],
    "settings": {
        "fontFamilies": [
            {
                "family": "Lemonada",
                "src": "https://fonts.googleapis.com/css2?family=Lemonada:wght@300&display=swap"
            }
        ],
        "theme": {
            "general": {
                "color": "#333333",
                "backgroundColor": "#ffffff",
                "borderColor": "#e6e6e6"
            },
            "overlay": {
                "backgroundColor": "rgba(255, 255, 255, 0.75)",
                "borderColor": "#dddddd",
                "boxShadow": "0 14px 28px rgba(0,0,0,0.25), 0 10px 10px rgba(0,0,0,0.22)",
                "color": "#333333"
            }
        }
    },
    ...
```

Pros

- easy to access from the state
- the list of font is related to a specific story

Cons

- the list is saved always in the story configuration
- update on newgeostory.json will affect only new created stories

in the future we could think to add a UI interaction to update the list of font families, to mitigate the cons of this approach 

please @tdipisa @offtherailz could you give me a feedback on this?

Reply:
@allyoucanmap thanks for the review. PR #5782 needs to merge first.

Reply:
@AGMETEOR I merged the master on this branch locally to test it.
Before proceed I'll add a comment with the expected configuration that I discussed with @offtherailz and @tdipisa this morning.

Reply:
This should be the expected configuration:

- list of font families should be listed in the localConfig.json inside the **GeoStory** plugin. This list should **replace** the existing list, for this reason we should take into account also the representation of safe webfont

```js
{
    "name": "GeoStory",
    "cfg": {
        "fontFamilies": [
            {
                "family": "Georgia",
                "safe": true
            },
            {
                "family": "Lemonada",
                "src": "https://fonts.googleapis.com/css2?family=Lemonada:wght@300&display=swap"
            }
        ]
    }
}
```
the list in the example above should load the font only for the one with src

- when the GeoStory plugin mounts it should perform following actions
  - on GeoStory mount: merge the fontFamilies props provided by localConfig with the fontFamilies saved in the theme settings of a story (if they exists)
  - on GeoStory mount: from the merged list of fontFamilies we should load the ones with an src (not safe). Ensure that the tags link are not loaded multiple times if they are already loaded in the header.
  - on GeoStory mount: store the merged fontFamilies in the geostory redux state to make them accessible from the GeoStoryEditor plugin

- we need to store the list of merged font families in the geostory configuration when a user save the story. We should add this information inside the `settings.theme.fontFamilies` path.

- if both the localConfig.json and the settings.theme.fontFamilies don't provide a list of fonts we should use the default ones

please take into account also the issues in the previous comment https://github.com/geosolutions-it/MapStore2/pull/5797#discussion_r481940499

Reply:
We discussed and the previous comment cover the needed use cases
@AGMETEOR  any questions?

Reply:
None atm